crates/biome_css_analyze/src/lint/nursery/no_invalid_position_at_import_rule.rs

```diff
@@ -0,0 +1,99 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_css_syntax::{AnyCssRule, CssRuleList};
+use biome_rowan::{AstNode, TextRange};
+
+declare_rule! {
+    /// Disallow the use of `@import` at-rules in invalid positions.
+    ///
+    /// Any `@import` rules must precede all other valid at-rules and style rules in a stylesheet (ignoring `@charset` and `@layer`), or else the `@import` rule is invalid.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```css,expect_diagnostic
+    /// a {}
+    /// @import 'foo.css';
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```css
+    /// @import 'foo.css';
+    /// a {}
+    /// ```
+    ///
+    pub NoInvalidPositionAtImportRule {
+        version: "next",
+        name: "noInvalidPositionAtImportRule",
+        recommended: true,
+        sources: &[RuleSource::Stylelint("no-invalid-position-at-import-rule")],
+    }
+}
+
+pub struct RuleState {
+    span: TextRange,
+}
```

Comment:
If you have struct with one field, the struct is superfluous. Just TextRange as state is enough

Reply:
fix: https://github.com/biomejs/biome/pull/2717/commits/f53625151b7d4f09aabe35e574aeadc39a2b3ce6